src/components/FilterBox/FilterBox.js

```diff
@@ -0,0 +1,156 @@
+import { useData } from "../../contexts";
+import { getAllLabels } from "../../utils/helperFunctions";
+import { MdKeyboardArrowDown, MdKeyboardArrowUp } from "react-icons/md";
+import "./FilterBox.css";
+
+export const FilterBox = ({
+  notes,
+  showFilters,
+  setShowFilters,
+  selectedFilters,
+  setSelectedFilters,
+}) => {
+  const colors = {
+    white: "white",
+    red: "red",
+    green: "green",
+    yellow: "yellow",
+  };
+
+  const tagList = getAllLabels(notes);
+
+  const isPresentInList = (itemToCheck, list) => {
+    return list.some((item) => item === itemToCheck);
+  };
+
+  const toggleTagHandler = (tag) => {
+    if (isPresentInList(tag, selectedFilters.selectedTags)) {
+      setSelectedFilters({
+        ...selectedFilters,
+        selectedTags: selectedFilters.selectedTags.filter(
+          (item) => item !== tag
+        ),
+      });
+    } else {
+      setSelectedFilters({
+        ...selectedFilters,
+        selectedTags: selectedFilters.selectedTags.concat(tag),
+      });
+    }
+  };
+
+  const toggleColorHandler = (color) => {
```

Comment:
Code is well structured and moduler. Good use of a color toggler here.

Reply:
Thanks Megha!